### ISSUE TITLE
Flight mode when reaching levels orange and above

### ISSUE THREAD

**Issue by explorethepaths** [SecUpwN/Android-IMSI-Catcher-Detector#340]
I think it will come in handy at times. I'm sorry for no labels, I'm new.

**Comment by explorethepaths** [comment]
I'm sorry for no labels, I'm new.

**Comment by SecUpwN** [comment]
Good morning and welcome to our project, @explorethepaths! Don't worry, you cannot add labels in our repo, this will always be done through someone who has reviewed your Issue. I have re-opened it since you closed it yourself (likely by accident). Let#s wait until @E3V3A tells us what he thinks about this.

**Comment by tobykurien** [comment]
Sadly, it's not possible for our app to turn on airplane mode, unless it is installed as a system app.

**Comment by E3V3A** [comment]
Actually [WRITE_SETTINGS](http://developer.android.com/reference/android/Manifest.permission.html#WRITE_SETTINGS) is no longer used ( API < 17) for Setting Airplane mode. The setting required today is the AOS non-3rd-party setting [WRITE_SECURE_SETTINGS](http://developer.android.com/reference/android/Manifest.permission.html#WRITE_SECURE_SETTINGS) (API  >= 17 or AOS 4.2.x )

So for devices using API <17 we can use the API, since it is not listed as a system permission (and AS doesn't complain), and we have it added into our manifest. 

There are several other ways we can do this, since we already have SU as a requirement. Two ways are:
 1. `su -c "am broadcast -a android.intent.action.AIRPLANE_MODE --ez state true"`
 2. `su -c "settings put global airplane_mode_on"`

But we can probably also use `su -c "setprop something 1"`. Or just make our app a system app. 

See: 
[1] [toggle-airplane-mode-in-android](https://stackoverflow.com/questions/5533881/toggle-airplane-mode-in-android)
[2] [how-to-turn-on-off-airplane-mode-even-on-new-android-versions-and-even-with-ro](http://stackoverflow.com/questions/25674655/how-to-turn-on-off-airplane-mode-even-on-new-android-versions-and-even-with-ro)

**Comment by SecUpwN** [comment]
Sounds like making our App a system App will prevent further struggle with other things in the future. But then again, I don't know if that would prevent it working correctly on non-rooted devices? Would like to hear @tobykurien on this since he has the deeper insight of which would be the best way to travel.

**Comment by tobykurien** [comment]
It would be great if it could be a system app on something like say the Blackphone. Perhaps that would enable more functionality. However, we should probably stick to our guns about just being a purely "detect" app, and leave the "responding/defense" to another app. So, what we could do is send out a system broadcast when our detection levels change, thus allowing other apps to respond as they want to. This also helps us focus on the important core functionality of detection, rather than get bogged down in irrelevant features.

**Comment by SecUpwN** [comment]
>... thus allowing other apps to respond as they want to...

Thanks for the hint, @tobykurien. Of course we're here to do detection for the most part, but beside triggering a notification to other Apps that may be installed, I still consider it to be useful for our own App to tackle countermeasures that will work on a broad range of devices. Of course we don't have to add stuff that does not work, but basic things (like acoustic warning as discussed in #15) and orther things we can enable on abroad range of devices that actually protect the user from further attacks should indeed be added.

**Comment by E3V3A** [comment]
@tobykurien I totally agree on your general design opinion. But I think being able to turn off WiFi and shut down mobile network functionality, is almost as essential as being able to close the App itself. Yeah, we could pop-up a message to the user saying "Warning: turn your phone off naaaow!", but how fun is that when you're sleeping or got your teddy in your pocket. Only the US and UK government spy-lobby would know why Google would choose to prevent people to turn their devices automatically to airplane mode. We should do whatever we can to be able to do this. 

Also, in case you've missed it, WiFi is always on and scanning the network (while giving away its location and MAC) even when you're WiFi indicator is off, and you think your WiFi is off. Nope! This is the reason why Chainfire developed [Pry-Fi](http://forum.xda-developers.com/showthread.php?t=2631512). 

There are many other options for shutting down radio, such as AT commands, root command-line and service mode injection, in case Google API ****-head developers doesn't wanna cooperate.

**Comment by banjaxbanjo** [comment]
This can be achieved using AT+CFUN to turn off radio but phone has to be rooted as E3V3A said.

**Comment by SecUpwN** [comment]
>This can be achieved using AT+CFUN to turn off radio but phone has to be rooted as E3V3A said.

@banjaxbanjo, I am wondering why our app only requests root when pressing the `AT Command Interface` button. Wouldn't it be better to detect it right after the start and when root access has been granted, let the user enable a new checkbox called `Flight Mode on Attack`, which would be located in `Preferences/PROTECTION SETTINGS` right above the Femtocell options? Without root, that checkbox should be grayed out.

**Comment by banjaxbanjo** [comment]
@SecUpwN it only requests root when that button is pressed because its using su user to access root settings.

The AT bit is a bit tricky to implement  because first you will needed to programmatically find an open serial port that accepts AT COMMANDS.
